### PR TITLE
[FEAT] Access Marketplace From Anywhere

### PR DESCRIPTION
**Pull request by spencerdezartsmith**
# Description

This PR makes the marketplace available from anywhere. The icon will appear in both the home hud and the world hud.

Fixes #issue

# What needs to be tested by the reviewer?

- Access the marketplace from anywhere
- Go back to where you came from when closing
- Maintain your bumpkin position in the mmo after returning from the marketplace

# Checklist:

- [x] Title of the PR is relevant and is prefixed with [FEAT], [CHORE] or [FIX]

### CODE REVIEW

File: src/features/marketplace/components/Collection.tsx [hunk 1]
```diff
@@ -4,7 +4,7 @@ import { loadMarketplace as loadMarketplace } from "../actions/loadMarketplace";
 import * as Auth from "features/auth/lib/Provider";
 import { useActor } from "@xstate/react";
 import { useNavigate, useSearchParams } from "react-router-dom";
-import { ListViewCard } from "./ListViewCard";
+import { useLocation, ListViewCard } from "./ListViewCard";
```
Comment (adamhannigan) [inline src/features/marketplace/components/Collection.tsx:7]: This is breaking the app. Small typo,`useLocation` should be imported from just above in `react-router-dom`

**Review by adamhannigan — COMMENTED** [review]
Nice work mate!

I just found a few small issues when closing the marketplace when I was a few levels deep. When I closed from the Home Page, everything worked well. However, when I navigated in the marketplace and then clicked the close button (cross icon in top right) I experienced strange behaviour.

1. Open marketplace
2. Navigate to "My trades"
3. Click "X" to close marketplace - takes me back to home page instead of the game

Additionally, if I do the above ^ in the plaza it black screens me when I do close the marketplace.

File: src/features/marketplace/components/Collection.tsx (hunk 1 above)
Comment (spencerdezartsmith) [reply src/features/marketplace/components/Collection.tsx:7]: Classic

**Review by adamhannigan — APPROVED** [review]
LGTM

**Pull request merged**